doc/dev/libraries.md

```diff
@@ -0,0 +1,84 @@
+- `libelektra-core`:
+  The core minimal API of Elektra.
+  Defines what `ElektraKey` and `ElektraKeyset` are, and contains the minimal API for manipulating them.
+
+  **Prefixes:** `elektraKey*` and `elektraKeyset*`
+
+- `libelektra-kdb`:
+  The main API for interfacing with the KDB.
+
+  **Prefix:** `elektraKdb*`
+
+- `libelektra-lowlevel-c`:
+  Additional C APIs that are useful when working with `ElektraKey` and `ElektraKeyset`, but not are not minimal API.
+  Specifically targets C and not intended for use via bindings.
+
+  **Prefix:** `elektra*`
+
+- `libelektra-highlevel-c`:
+  The C high-level API for reading/writing configuration with Elektra.
+  Intended for use by applications.
+  Partially intended for use with code-generation.
+  Specifically targets C and not intended for use via bindings.
+
+  **Prefix:** `elektraHlc*` (better suggestions welcome)
+  (Note: Even though there isn't any other `highlevel` implementation, I added the `-c` suffix to show this lib is only for C)
+
+- `libelektra-opts`:
+  The API for parsing command-line arguments according to Elektra's spec.
+  **Prefix:** `elektraOpts*`
+- `libelektra-notification`:
+  The API for setting up callbacks and automatically updated variables linked to changes in the KDB.
+
+  **Prefix:** `elektraNotification*`
+
+- `libelektra-io`:
+  Elektra's API for asynchronous operations.
+
+  **Prefix:** `elektraIo*`
+
+- `libelektra-merge`:
+  The API for merging and/or detecting conflicts between two keysets.
+
+  **Prefix:** `elektraMerge`
+
+- `libelektra-pluginbase`:
+  The API _used by_ plugins.
+
+  **Prefix:** `elektraPlugin*`
+  (Note: previously `libelektra-plugin`)
+
+- `libelektra-plugin`:
+  The API for _interacting with_ plugins.
+  This includes loading plugins and calling exported functions.
+
+  **Prefix:**: `elektraPlugin*`
+  (Note: previously `libelektra-invoke` & `loader`)
```

Comment:
loader was part of kdb. Do you want to use the invoke lib within kdb?

Reply:
I wasn't sure about that part. Either we depend on this library from `libelektra-kdb`, or we build the 1 or 2 loader functions into a static library that is linked into both `libelektra-kdb` and this library. Since the loader library should be very small in the `BUILD_SHARED` variant, I think this duplication would be acceptable. In the `BUILD_STATIC` and `BUILD_FULL` variants everything is bundled into one library anyway.

Reply:
I agree, the duplication makes sense in this case.